cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -184,7 +185,42 @@ func generateMountsModeRecover() error {
 }
 
 func generateMountsModeRun() error {
-	return fmt.Errorf("run mode mount generation not implemented yet")
+	seedDir := filepath.Join(runMnt, "ubuntu-seed")
+	bootDir := filepath.Join(runMnt, "ubuntu-boot")
+	dataDir := filepath.Join(runMnt, "ubuntu-data")
+
+	// 1. always ensure basic partitions are mounted
+	for _, d := range []string{seedDir, bootDir, dataDir} {
+		isMounted, err := osutilIsMounted(d)
+		if err != nil {
+			return err
+		}
+		if !isMounted {
+			fmt.Fprintf(stdout, "/dev/disk/by-label/%s %s\n", filepath.Base(d), d)
```

Comment:
Are we always safe against spaces here? Is there a need for a typical space-escape handling like various osutil mount-related parsing/generating functions?

Reply:
Currently yes, the partition names are all very strictly defined, i.e. ubuntu-{seed,boot,data}. If we open this up later we could enforce no-spaces via the gadget validations.